src/main/src/main/java/org/geoserver/config/impl/ContactInfoImpl.java

```diff
@@ -50,6 +82,8 @@ public void setId(String id) {
 
     @Override
     public String getAddress() {
+        if (address == null && internationalAddress != null)
```

Comment:
I'm not sure I'm following the logic, but why do we need the `address == null` part of the test? Shouldn't we just prefer `internationalAddress` if its available?

[This is a general comment about the implementation, but the response will be same for all of these cases, so just putting it here]

Reply:
in theory the two are mutually exclusive. However from UI the normal string field is set to null only when saving (not when pressing apply). The additional null check is meant for this case.